### PR TITLE
Included task to check status of disk pool pods

### PR DESCRIPTION
**Pull request by shashank855**
**What this PR does / why we need it**:
- Included task to check status of disk pool pods

### CODE REVIEW

File: providers/cstor/cstor-disk-pool/test.yml [hunk 1]
```diff
@@ -48,12 +65,11 @@
               args:
                 executable: /bin/bash
 
-            - name: Get the maxpool count of storage pool claim.
-              shell: kubectl get spc -o=jsonpath='{.items[?(@.metadata.name=="{{pool_name}}")].spec.maxPools}'
+            - name: Getting the number of Disks
```
Comment (gprasath) [inline providers/cstor/cstor-disk-pool/test.yml]: nodes

File: providers/cstor/cstor-disk-pool/test.yml [hunk 2]
```diff
@@ -22,12 +22,29 @@
...
-          shell: kubectl get disks -l ndm.io/disk-type=disk --no-headers -o custom-columns=:metadata.name
+        - name: Getting Used disks
+          shell: kubectl get sp -o jsonpath='{.items[*].spec.disks.diskList}' | tr " " "\n" |sed 's/[][]//g'
           args:
             executable: /bin/bash
-          register: diskList
-          failed_when: "'disk-' not in diskList.stdout"
+          register: used_disks
+      
+        - local_action: copy content={{ used_disks.stdout }} dest="./used_disks"
```
Comment (nsathyaseelan) [inline providers/cstor/cstor-disk-pool/test.yml]: :+1:

File: providers/cstor/cstor-disk-pool/test.yml [hunk 3]
```diff
@@ -62,7 +78,7 @@
               args:
                 executable: /bin/bash
               register: pool_count
-              until: pool_count.stdout_lines|length == maxpool_count.stdout|int
+              until: pool_count.stdout_lines|length == diskList.stdout_lines|length
```
Comment (dargasudarshan) [inline providers/cstor/cstor-disk-pool/test.yml]: @shashank855 Does this validation works for Mirrored pool?

File: providers/cstor/cstor-disk-pool/test.yml [hunk 4]
```diff
@@ -22,12 +22,29 @@
...
+        - local_action: copy content={{ all_disks.stdout }} dest="./all_disks"
+
+        - name: Copying content
+          shell: cat all_disks used_disks >> tmp_disks
+
+        - name: Getting list of unused disks attached
+          shell: |
+            sort used_disks > a1
+            sort all_disks > a2   
+            diff a2 a1 | grep -e "<" -e ">" | awk '{print $2}'
```
Comment (ksatchit) [inline providers/cstor/cstor-disk-pool/test.yml]: Not sure if this is a relevant review comment! can we keep all this logic separate somewhere - improved NDM operators should be capable of updating attributes in the disk CR on whether it is clamined (part of pool) or no. In which case, the util can be updated and the test can remain untouched ?

File: providers/cstor/cstor-disk-pool/test.yml (hunk 3 above)
Comment (shashank855) [reply providers/cstor/cstor-disk-pool/test.yml]: @dargasudarshan No for mirrored pool separate util has to be written.

File: providers/cstor/cstor-disk-pool/test.yml (hunk 4 above)
Comment (shashank855) [reply providers/cstor/cstor-disk-pool/test.yml]: @ksatchit SP is going to deprecate in near future as discussed with dev-team so the entire picture of our util for finding the unused disk is going to change. This PR is temporary workaround once we will have proper labels in our disk CRs we will write a separate util for this purpose.

Comment (ksatchit) [reply providers/cstor/cstor-disk-pool/test.yml]: Thanks @shashank855

**Pull request merged**